File: src/main/java/de/moritzf/quota/idea/QuotaSettingsListener.java
package de.moritzf.quota.idea;

import com.intellij.util.messages.Topic;

/**
 * Broadcast when plugin settings affecting UI rendering have changed.
 */
public interface QuotaSettingsListener {
    Topic<QuotaSettingsListener> TOPIC =
            Topic.create("openai.usage.quota.settings", QuotaSettingsListener.class);

    void onSettingsChanged();
}

File: src/main/kotlin/de/moritzf/quota/idea/QuotaSettingsListener.kt
package de.moritzf.quota.idea

import com.intellij.util.messages.Topic

/**
 * Broadcast when plugin settings affecting UI rendering have changed.
 */
fun interface QuotaSettingsListener {
    fun onSettingsChanged()

    companion object {
        @JvmField
        val TOPIC: Topic<QuotaSettingsListener> = Topic.create("openai.usage.quota.settings", QuotaSettingsListener::class.java)
    }
}
